src/ansible_navigator/image_manager/puller.py

```diff
@@ -142,18 +151,33 @@ def prologue_stdout(self):
         print("Updating the execution environment")
         print("\u002d" * width)
 
+    def _generate_pull_command(self) -> List:
+        """Generate the pull command.
+
+        :returns: The list of command parts
+        """
+        command_line = [self._container_engine, "pull"]
+        if isinstance(self._arguments, list):
+            for argument in self._arguments:
+                command_line.extend(shlex.split(argument))
+        command_line.append(self._image)
+        return command_line
+
     def pull_stdout(self):
         """pull the image, print to stdout
 
         podman writes to stderr
         docker writes to stdout
         """
         try:
+            command_line = self._generate_pull_command()
+            shlex_joined = " ".join(shlex.quote(arg) for arg in command_line)
```

Comment:
`shlex.join()`

Reply:
Also, is this necessary? I haven't followed the logic all the way so it might be fine, but we should make sure we're not double escaping.

Reply:
Ah okay it's just for rendering, not for `subprocess.run()` below. Carry on, then. :)